modules/apps/portal-search/portal-search-admin-web/src/main/resources/META-INF/resources/js/main.js

```diff
@@ -97,6 +102,147 @@ AUI.add(
+						const totalDiskSpace =
+							availableDiskSpace + currentDiskSpaceUsed;
+
+						const usedPercentage =
+							(currentDiskSpaceUsed /
+								(availableDiskSpace + currentDiskSpaceUsed)) *
+							100;
+
+						const progressBarHTML =
+							'<label>' +
+							Liferay.Language.get('disk-usage') +
+							'</label>' +
+							'<div class="progress ' +
+							(isLowOnDiskSpace && 'progress-warning') +
+							'">' +
+							'<div aria-valuemax="100" aria-valuemin="0" aria-valuenow="' +
+							usedPercentage +
+							'" class="progress-bar" role="progressbar" style="width: ' +
+							usedPercentage +
+							'%;"></div>' +
+							'</div>' +
+							'<div class="text-3 text-secondary">' +
+							'<span>' +
+							Liferay.Util.sub(
+								Liferay.Language.get('used-x-of-x-gb'),
+								currentDiskSpaceUsed.toFixed(1),
+								totalDiskSpace.toFixed(1)
+							) +
+							'</span>' +
+							'<span class="float-right">' +
+							Liferay.Util.sub(
+								Liferay.Language.get('x-gb-free'),
+								availableDiskSpace.toFixed(1)
+							) +
+							'</span>' +
+							'</div>';
+
+						const lowDiskSpaceDescriptionHTML =
+							'<span>' +
+							Liferay.Language.get(
+								'reindex-elasticsearch-disk-space-warning'
+							) +
+							':' +
+							'</span>' +
+							'<ul>' +
+							'<li>' +
+							Liferay.Util.sub(
+								Liferay.Language.get('available-disk-space-x'),
+								availableDiskSpace.toFixed(1)
+							) +
+							'</li>' +
+							'<li>' +
+							Liferay.Util.sub(
+								Liferay.Language.get(
+									'current-disk-space-used-x'
+								),
+								currentDiskSpaceUsed.toFixed(1)
+							) +
+							'</li>' +
+							'</ul>' +
+							'<span class="c-mt-2">' +
+							Liferay.Language.get(
+								'do-you-still-wish-to-execute-reindex'
+							) +
+							'</span>';
+
+						bodyHTML = isLowOnDiskSpace
+							? progressBarHTML +
+							  '<div class="text-secondary c-mt-2">' +
+							  lowDiskSpaceDescriptionHTML +
+							  '</div>'
+							: infoHTML +
+							  '<div class="panel" role="tablist">' +
+							  '<button aria-controls="collapsePanel" aria-expanded="false" class="disk-space-panel collapsed btn btn-link c-pl-0 c-pb-0" data-target="#collapsableDiskSpace" data-toggle="liferay-collapse" role="tab">' +
+							  '<span class="collapse-button-closed">' +
+							  Liferay.Language.get('view-disk-space') +
+							  '</span>' +
+							  '<span class="collapse-button-open">' +
+							  Liferay.Language.get('hide-disk-space') +
+							  '</span>' +
+							  '</button>' +
+							  '<div class="panel-collapse collapse" id="collapsableDiskSpace" role="tabpanel">' +
+							  '<div class="panel-body c-pl-0 c-pr-0">' +
+							  progressBarHTML +
+							  '</div>' +
+							  '</div>' +
+							  '</div>' +
+							  checkboxHTML;
+					}
+
+					return (
+						'<div class="reindex-actions-confirmation-modal-body">' +
+						bodyHTML +
+						'</div>'
+					);
```

Comment:
Yea... it has been a challenge trying to get the modal to do exactly what's expected without React :( Do you mean rewriting all of `main.js` into a React component? I hadn't thought of that... I assumed it would take a while to figure out what the other parts of main was doing?

Reply:
Ya, re-writing all of `main.js`, essentially making the entire page a react app. Re-writing means keeping the same functionality, not necessarily meaning to keep the same logic; in fact I don't think much of the logic should stay the same after coding in a react way.

We can create another task for the re-write now that this is functional already, but I'd say we should definitely do it soon to make future changes easier to maintain or debug.